### ISSUE TITLE
Class types don't satisfy callable Protocols with ParamSpec

### ISSUE THREAD

**Issue by chasleslr** [astral-sh/ty#3117]
## Summary

ty incorrectly rejects class types as arguments to callable Protocols that use `ParamSpec`, even when the class's `__init__` signature matches the Protocol's `__call__` signature.

Both pyright (1.1.408) and mypy (1.19.1 --strict) accept this code with no errors.

## Reproduction

```python
from typing import ParamSpec, Protocol

P = ParamSpec("P")


class App(Protocol):
    """Represents an ASGI app."""
    async def __call__(self, scope: dict[str, object], receive: object, send: object) -> None: ...


class MiddlewareFactory(Protocol[P]):
    """
    A factory that wraps an ASGI app with middleware.
    This is equivalent to Starlette's _MiddlewareFactory.
    """
    def __call__(self, app: App, /, *args: P.args, **kwargs: P.kwargs) -> App: ...


class ConcreteMiddleware:
    """
    A middleware class. Calling ConcreteMiddleware(app, option="x") returns
    a ConcreteMiddleware instance, which satisfies App (has __call__).
    """
    def __init__(self, app: App, /, option: str = "default") -> None:
        self.app = app
        self.option = option

    async def __call__(self, scope: dict[str, object], receive: object, send: object) -> None:
        await self.app(scope, receive, send)


def add_middleware(
    middleware_class: MiddlewareFactory[P],
    *args: P.args,
    **kwargs: P.kwargs,
) -> None:
    pass


# ERROR: Expected `MiddlewareFactory[(...)]`, found `<class 'ConcreteMiddleware'>`
add_middleware(ConcreteMiddleware, option="test")


# For comparison: the same pattern WITHOUT ParamSpec works correctly
class SimpleFactory(Protocol):
    def __call__(self, app: App, /) -> App: ...

class SimpleMiddleware:
    def __init__(self, app: App, /) -> None:
        self.app = app
    async def __call__(self, scope: dict[str, object], receive: object, send: object) -> None:
        await self.app(scope, receive, send)

def add_simple_middleware(middleware_class: SimpleFactory) -> None:
    pass

# This PASSES
add_simple_middleware(SimpleMiddleware)
```

**Expected:** All checks pass  
**Actual:** ty reports `invalid-argument-type` on line 42

## Analysis

The class `ConcreteMiddleware` should satisfy `MiddlewareFactory[P]` because:
- `ConcreteMiddleware.__init__` signature: `(self, app: App, /, option: str = "default") -> None`
- `MiddlewareFactory.__call__` expects: `(self, app: App, /, *args: P.args, **kwargs: P.kwargs) -> App`
- Calling `ConcreteMiddleware(app, option="x")` returns a `ConcreteMiddleware` instance
- `ConcreteMiddleware` instances satisfy `App` (they have compatible `__call__`)

The same pattern **without** `ParamSpec` works correctly (see `SimpleFactory` example above).

## Real-world Impact

This breaks type checking for **Starlette/FastAPI middleware registration**:

```python
from fastapi import FastAPI
from starlette.middleware.cors import CORSMiddleware

app = FastAPI()
app.add_middleware(CORSMiddleware, allow_origins=["*"])  # ty rejects this
```

Starlette defines [`_MiddlewareFactory`](https://github.com/encode/starlette/blob/master/starlette/middleware/__init__.py) as a Protocol with ParamSpec, and `add_middleware` expects this type.

## Other Type Checkers

| Type Checker | Version | Result |
|--------------|---------|--------|
| pyright | 1.1.408 | ✅ 0 errors |
| mypy --strict | 1.19.1 | ✅ 0 errors |
| ty | 0.0.24 | ❌ 1 error |

## Environment

- ty version: 0.0.24
- Python: 3.13
- OS: macOS

**Comment by carljm** [comment]
Hi, thanks for the report! This looks like a duplicate of #1635

**Comment by dhruvmanila** [comment]
This has been fixed on main: [Playground](https://play.ty.dev/621a4457-50f2-481b-be11-3ee3fa22eb6b) and it should be available in the next release.
